source/common/config/new_grpc_mux_impl.cc

```diff
@@ -69,7 +70,9 @@ void NewGrpcMuxImpl::onDiscoveryResponse(
   }
 
   kickOffAck(sub->second->sub_state_.handleResponse(*message));
-  Memory::Utils::tryShrinkHeap();
+  auto shrink_threshold = runtime_.threadsafeSnapshot()->getInteger(
+      "envoy.memory.heap_shrink_threshold", static_cast<uint64_t>(100 * 1024 * 1024UL));
+  Memory::Utils::tryShrinkHeap(shrink_threshold);
```

Comment:
Why is this not just done inside the function? I think that makes more sense?

Reply:
There should be layer without the dependency of runtime.

In the pazst Memory::Utils::tryShrinkHeap is the one because it's static function and can be used anywhere. 

I am open to create a util instance passing the runtime arg and hide the runtime key there. 

Should I do it?

Reply:
We allow static access to runtime as a singleton. There is always a runtime, even if it is "null". In that case you can just return the default value.

Reply:
Thank you for the explanation! Let me try